src/multi_stage/configure_multi_stage_inputs.jl

```diff
@@ -101,6 +121,17 @@ function configure_multi_stage_inputs(inputs_d::Dict, settings_d::Dict, NetworkE
 	inputs_d["RET_CAP_ENERGY"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["STOR_ALL"])
 	# Set of asymmetric charge/discharge storage resources eligible for charge capacity retirements
 	inputs_d["RET_CAP_CHARGE"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["STOR_ASYMMETRIC"])
+	# Set of all co-located resources' components eligible for capacity retirements
+	if !isempty(inputs_d["VRE_STOR"])
+		inputs_d["RET_CAP_DC"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["VS_DC"])
+		inputs_d["RET_CAP_SOLAR"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["VS_SOLAR"])
+		inputs_d["RET_CAP_WIND"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["VS_WIND"])
+		inputs_d["RET_CAP_STOR"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["VS_STOR"])
+		inputs_d["RET_CAP_DISCHARGE_DC"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["VS_ASYM_DC_DISCHARGE"])
+		inputs_d["RET_CAP_CHARGE_DC"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["VS_ASYM_DC_CHARGE"])
+		inputs_d["RET_CAP_DISCHARGE_AC"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["VS_ASYM_AC_DISCHARGE"])
+		inputs_d["RET_CAP_CHARGE_AC"] = intersect(dfGen[dfGen.New_Build.!=-1,:R_ID], inputs_d["VS_ASYM_AC_CHARGE"])
```

Comment:
suggest defining a 'set' or vector here which is `dfGen[dfGen.New_Build.!=-1,:R_ID]` since it gets used 8 times